src/tests/JIT/opt/Devirtualization/Comparer_get_Default.cs

```diff
@@ -252,6 +423,20 @@ public struct Struct1 : IComparable
     public long b;
     public int CompareTo(object obj)
     {
-        return b.CompareTo(((Struct1) obj).b);
+        return obj is Struct1 str ?  b.CompareTo(str.b) : 1;
     }
 }
+
+public struct Struct2
+{
+    public long a;
+    public long b;
+}
+
+public struct StructGeneric<T> : IEquatable<StructGeneric<T>>, IComparable<StructGeneric<T>>
```

Comment:
Another interesting case is `struct G<T> : IEquatable<string>`.

`getDefaultEqualityComparerClassHelper on `struct G<Canon>` needs to return null for this case since it cannot tell whether Canon is going to be string or not.